gossip/src/cluster_info.rs

```diff
@@ -2444,7 +2441,9 @@ impl ClusterInfo {
         self.stats
             .packets_sent_push_messages_count
             .add_relaxed((packet_batch.len() - num_prune_packets) as u64);
-        let _ = response_sender.send(packet_batch);
+        if !packet_batch.is_empty() {
```

Comment:
Should all the stats counter increases be surrounded by this if as well?

Reply:
hmmm, not sure it matters much either way:
* the stats counters are just doing an atomic add, and even if the argument is zero it wouldn't really have a measurable impact.
* except for the prune counters, I would expect the other counters to be almost always non-zero anyways. i.e. we always push some messages out when receiving push messages.

Given that, I lean towards keeping the `if` branch shorter and simpler here. But if we decide to change that, because this is a backport, I would rather to do so on master branch first otherwise v2.0 becomes inconsistent with master.

Reply:
assuming the question was asked wrt the metrics remaining accurate with this change?

Reply:
The metrics are accurate.
My understanding was that the comment was about avoiding `metric.add_relaxed(0)` which is a no-op.

Reply:
Yep, trying to avoid some wasted time on no-ops and make it match ping handling a little more closely https://github.com/anza-xyz/agave/blob/master/gossip/src/cluster_info.rs#L2561-L2571

But I don't see anything functionally wrong, and agree this isn't turning the needle on performance in any meaningful way.